ci-operator/step-registry/p2p/acm-cluster-install/p2p-acm-cluster-install-commands.sh

```diff
@@ -0,0 +1,343 @@
+   namespace: ${CLUSTER_NAME}
+   labels:
+    cloud: 'AWS'
+    region: '${REGION}'
+    vendor: OpenShift
+    cluster.open-cluster-management.io/clusterset: '${CLUSTER_NAME}-set'
+   annotations:
+    hive.openshift.io/cluster-lifespan: 6h
+spec:
+   baseDomain: ${BASE_DOMAIN}
+   clusterName: ${CLUSTER_NAME}
+   controlPlaneConfig:
+    servingCertificates: {}
+   platform:
+      aws:
+         region: ${REGION}
+         credentialsSecretRef:
+            name: acm-aws-secret
+   pullSecretRef:
+      name: pull-secret
+   installAttemptsLimit: 1
+   provisioning:
+      installConfigSecretRef:
+         name: install-config
+      releaseImage: ${CLUSTER_IMAGE}
+      sshPrivateKeyRef:
+         name: ssh-private-key
+EOF
+
+oc apply -f "$CLUSTER_DEPLOYMENT_FILE"
+
+# create managed cluster reosurce
+MANAGED_CLUSTER_FILE=/tmp/managed_cluster.yaml
+cat > $MANAGED_CLUSTER_FILE <<EOF
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: $CLUSTER_NAME
+  labels:
+    name: $CLUSTER_NAME
+    cloud: Amazon
+    region: $REGION
+    vendor: OpenShift
+    cluster.open-cluster-management.io/clusterset: ${CLUSTER_NAME}-set
+spec:
+  hubAcceptsClient: true
+EOF
+
+oc apply -f "$MANAGED_CLUSTER_FILE"
+
+# create klusterlet addon config
+KLUSTERLET_ADD_ON_CONFIG_FILE=/tmp/klusterletaddonconfig.yaml
+cat > $KLUSTERLET_ADD_ON_CONFIG_FILE <<EOF
+apiVersion: agent.open-cluster-management.io/v1
+kind: KlusterletAddonConfig
+metadata:
+  name: $CLUSTER_NAME
+  namespace: $CLUSTER_NAME
+spec:
+  clusterName: $CLUSTER_NAME
+  clusterNamespace: $CLUSTER_NAME
+  clusterLabels:
+    cloud: Amazon
+    vendor: OpenShift
+  applicationManager:
+    enabled: true
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: true
+  certPolicyController:
+    enabled: true
+EOF
+
+oc apply -f ${KLUSTERLET_ADD_ON_CONFIG_FILE}
+
+#=====================
+oc -n "$CLUSTER_NAME" patch clusterdeployment "$CLUSTER_NAME" --type=merge -p '{"spec":{"powerState":"Running"}}' >/dev/null || true
+
+oc -n "$CLUSTER_NAME" annotate clusterdeployment "$CLUSTER_NAME" hive.openshift.io/try-install-now=true --overwrite >/dev/null || true
+
+#=================================
+deadline=$(( $(date +%s) + INSTALL_TIMEOUT_MINUTES*60 ))
+# wait for cluster provision to show up
+echo "[INFO] $(now) Waiting for ClusterProvision to be created..."
+while true; do
+  cnt=$(oc -n "$CLUSTER_NAME" get clusterprovisions -l hive.openshift.io/cluster-deployment-name="$CLUSTER_NAME" --no-headers 2>/dev/null | wc -l | tr -d ' ')
+  [[ "$cnt" =~ ^[0-9]+$ ]] || cnt=0
+  if (( cnt > 0 )); then
+     echo "[INFO] ClusterProvision detected."
+     break
+  fi
+  if (( $(date +%s) > deadline )); then
+     echo "[ERROR] Timeout waiting for ClusterProvision"
+     exit 1
+  fi
+  sleep "$POLL_SECONDS"
+done
+
+#=====================
+STREAM_PID=""
+CURRENT_POD=""
+
+cleanup() {
+    [[ -n "${STREAM_PID:-}" ]] && kill "${STREAM_PID}" 2>/dev/null || true
```

Comment:
If you want to test a var. content, do not assign it any default value and I don't think you need `|| true`. Try do use syntax that is simpler and easier to maintain, unless it's really necessary.
```suggestion
    [[ -n "${STREAM_PID}" ]] && kill "${STREAM_PID}" 2>/dev/null
```

Reply:
This is must do this PR.